### DISCUSSION TITLE
Seek play position from http web server

### DISCUSSION THREAD

**Discussion by morfeus2** [schreibfaul1/ESP32-audioI2S#808] [General]
I think that seeking forward and reverse in audio coming from the network would be very useful (especially in long podcast or long recordings/music tracks). I tried the following without success:
audio.setTimeOffset()
audio.setAudioPlayPosition()
From the source I guess they are only for files on SD card.

I am thinking to files coming from web server, using "HTTP Range request header".

I don't see any other ways to obtain it in the current sources (maybe I missed something) and no discussions on this topic. Would it be possible to suggest to implement it?

**Comment by schreibfaul1** [comment]
Yes, there are ByteRange-Requests, but unfortunately this has not yet been implemented.

**Reply by morfeus2** [reply]
Do you have any plans to implement it in the future?
(I think that my skills in C++ are not enough to implement it by myself)

**Comment by schreibfaul1** [comment]
Yes, I have done some successful tests, it is possible, I just don't have the time at the moment. But that will come soon.

**Comment by morfeus2** [comment]
This is great, thank you!
I was thinking in these days, that when it will work, I have an idea to implement an automatic resume of the play position even across different "client devices" that use this library:
-when you stop the current play, audio.getAudioCurrentTime() retrives the current position and sends it via HTTP-GET to a php page that stores it on the server in a simple file
-when you restart the track, even from a different device, the client requests that file and if it's present and with valid data, seeks to that position before starting playing
I think I will be able to implement it by myself, if it's working and you are interested I can post here the sources.

**Reply by schreibfaul1** [reply]
The server always transmits the content length and probably also the bit rate. If the position where the audio data begins is also known, the metadata (such as images) can be calculated and the actual playback time determined. It would then be possible to jump to a specific time stamp.
